src/Room.java

```diff
@@ -26,22 +26,26 @@ void setCommoditySet(HashSet<AbstractCommodity> givenCommoditySet){
         this.commoditySet = givenCommoditySet;
     }
 
-    //Prepares commodities in a room & adds a maintenance record through string
-    void prepareCommodities(String date) throws ParseException {
-        //Iterator<AbstractCommodity> itr = this.commoditySet.iterator();
-        for (AbstractCommodity s : this.commoditySet) {
-            s.prepare();
+    Date stringToDate(String stringDate){
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy/mm/dd");
+        try {
+
+            Date date = formatter.parse(stringDate);
+            return date;
+
+        } catch (ParseException e) {
+            System.out.println("Parse throws exception");
+            return null;
```

Comment:
Never return null in a try/catch, you can throw a different exception with a message.